domain/application/service/application.go

```diff
@@ -84,7 +84,8 @@ type ApplicationState interface {
 
 	// GetUnitCloudContainerStatus returns the cloud container status of the specified
 	// unit, returning an error satisfying [applicationerrors.UnitNotFound] if the unit
-	// doesn't exist.
+	// doesn't exist, or an error satisfying [applicationerrors.UnitStatusNotFound] if
```

Comment:
It's not a change introduced in this PR, but I'm thinking that maybe `UnitStatusNotSet` was a better name for this error.

Reply:
Yeah probably, I'll follow up later